python/ray/data/datasource/tfrecords_datasource.py

```diff
@@ -64,15 +77,23 @@ def _write_block(
 
 def _convert_example_to_dict(
     example: "tf.train.Example",
+    tf_schema: Optional["schema_pb2.Schema"],
 ) -> Dict[str, "pyarrow.Array"]:
     record = {}
     for feature_name, feature in example.features.feature.items():
-        record[feature_name] = _get_feature_value(feature)
+        schema_feature_type = None
+        if tf_schema is not None:
+            for schema_feature in tf_schema.feature:
```

Comment:
nit: it might be more efficient to covert `tf_schema` into a dict first before looking up the feature type.